x/programs/rust/wasmlanche_sdk/src/program.rs

```diff
@@ -32,48 +41,37 @@ impl Program {
         target: &Program,
         max_units: i64,
         function_name: &str,
-        args: &[Box<dyn Argument>],
+        args: Vec<Vec<u8>>,
     ) -> i64 {
-        call_program(
-            self,
-            target,
-            max_units,
-            function_name,
-            marshal_args(args).as_ref(),
-        )
+        // flatten the args into a single byte vector
+        let args = args.into_iter().flatten().collect::<Vec<u8>>();
+        call_program(self, target, max_units, function_name, &args)
     }
 }
 
-impl From<Program> for i64 {
-    fn from(program: Program) -> Self {
-        program.id()
-    }
-}
+#[macro_export]
+macro_rules! params {
+    ($($param:expr),*) => {
+        {
+            // the macro expands into this block. This will serialize every parameter
+            // into a byte vector and return a vector of byte vectors.
+            let mut params = Vec::new();
+            $(
+                params.push(wasmlanche_sdk::program::serialize_params(&$param).unwrap());
+            )*
 
-impl From<i64> for Program {
-    fn from(value: i64) -> Self {
-        Self { id: value }
-    }
+            params
+        }
+    };
```

Comment:
This should be simplified (and publicly documented)
```rust
/// serialize every parameter into a byte vector and return a vector of byte vectors
#[macro_export]
macro_rules! params {
    ($($param:expr),*) => {
        vec![$(wasmlanche_sdk::program::serialize_params($param).unwrap(),)*]
    };
}
```
Also, notice I removed the `&`. It's repetitive for the caller to have to add the character, but then it makes `params!` have something more akin to function calling semantics, where it's clear that you can still use whatever was passed to the macro.